src/libmongoc/tests/test-mongoc-write-commands.c

```diff
@@ -626,6 +626,37 @@ _test_invalid_wc_server_error (void *unused)
    mongoc_client_destroy (client);
 }
 
+static void
+_test_command_error (void *unused)
+{
+   mongoc_client_t *client;
+   mongoc_collection_t *coll;
+   bool ret;
+   bson_t reply;
+   bson_error_t error;
+
+   client = test_framework_client_new ();
+   mongoc_client_set_error_api (client, MONGOC_ERROR_API_VERSION_2);
+   coll = get_test_collection (client, "server_wc_error");
+
+   _configure_failpoint (client,
+                         "{'times': 2}",
+                         "{'failCommands': ['insert'], 'errorCode' : 123 }");
+   ret = mongoc_collection_insert_one (
+      coll, tmp_bson ("{'x':1}"), NULL /* opts */, &reply, &error);
+   BSON_ASSERT (!ret);
+   ASSERT_ERROR_CONTAINS (error, MONGOC_ERROR_SERVER, 123, "");
+   BSON_ASSERT (bson_has_field (&reply, "code"));
+   BSON_ASSERT (bson_has_field (&reply, "codeName"));
+   BSON_ASSERT (bson_has_field (&reply, "errmsg"));
```

Comment:
Would it be too much trouble to ask that you also verify these fields will be available via the bulk API? See [PHPC-1386](https://jira.mongodb.org/browse/PHPC-1386) for how we originally came across this issue.

Reply:
Not at all! AFAIK all write results go through the same code path for constructing and merging write results, but it seems like a good idea to be sure.

I added a test for bulk.